pkg/cmd/pulumi/new_test.go

```diff
@@ -818,3 +818,49 @@ func TestErrorIfNotEmptyDirectory(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateStackRefAndProjectName(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		projectName string
+		stackRef    string
+		valid       bool
+	}{
+		{
+			projectName: "foo",
+			stackRef:    "foo",
+			valid:       true,
+		},
+		{
+			projectName: "fooo",
+			stackRef:    "org/foo/dev",
+			valid:       false,
+		},
+		{
+			projectName: "",
+			stackRef:    "org/foo/dev",
+			valid:       true,
+		},
+		{
+			projectName: "foo",
+			stackRef:    "",
+			valid:       true,
+		},
+		{
+			projectName: "foo",
+			stackRef:    "org/foo/dev",
+			valid:       true,
+		},
+	}
+	b, err := currentBackend(context.Background(), nil, display.Options{})
+	require.NoError(t, err)
+
+	for _, tt := range tests {
+		err := validateStackProjectName(b, tt.stackRef, tt.projectName)
+		if tt.valid {
+			assert.NoError(t, err, "projectName: %s, stackRef: %s", tt.projectName, tt.stackRef)
+		} else {
+			assert.Error(t, err, "projectName: %s, stackRef: %s", tt.projectName, tt.stackRef)
+		}
```

Comment:
optional: Instead of adding context to the assertions, you can put context in the subtest names.

```go
for _, tt := range tests {
  tt := tt
  t.Run(fmt.Sprintf("project=%q/stackRef=%q", tt.projectName, tt.stackRef), func(t *testing.T) {
    t.Parallel()

    // ...
  })
```